Hadrons/DistilMatrix.hpp

```diff
@@ -337,6 +560,118 @@ ::fetchDvBatchComponents(unsigned int ibatch, std::vector<unsigned int> time_dil
     return batch_dt;
 }
 
+// makePinned methods build distil vectors with reorganised time slices
+// (in order to compute multiple time-dilution blocks at different t with a single call of MesonField kernel)
+template <typename FImpl, typename T, typename Tio>
+void DmfComputation<FImpl,T,Tio>
+::makePinnedPhiComponent(FermionField&              phi_component,
+                   DistillationNoise&               n,
+                   const unsigned int               n_idx,
+                   const unsigned int               D,
+                   const unsigned int               delta_t,
+                   PerambTensor&                    peramb,
+                   LapPack&                         epack)
+{
+    std::array<unsigned int,3> d_coor = n.dilutionCoordinates(D);
+    unsigned int dt = d_coor[Index::t] , dk = d_coor[Index::l] , ds = d_coor[Index::s];
+    
+    //compute at pin_time, insert at t=pin_time
+    const unsigned int pin_time = (dt + delta_t)%nt_;               //TODO: generalise to dilution
+    const unsigned int t        = pin_time;                         //TODO: generalise to dilution 
+
+    std::vector<int> peramb_ts = peramb.MetaData.timeSources;
+    std::vector<int>::iterator itr_dt = std::find(peramb_ts.begin(), peramb_ts.end(), dt);
+    unsigned int idt_peramb = std::distance(peramb_ts.begin(), itr_dt); //gets correspondent index of dt in the tensor obj 
+    const unsigned int nVec = epack.evec.size();
+    const unsigned int Nt_first = g_->LocalStarts()[nd_ - 1];
+    const unsigned int Nt_local = g_->LocalDimensions()[nd_ - 1];
+    if( (pin_time>=Nt_first) and (pin_time<Nt_first+Nt_local) )
```

Comment:
this is where it would be great if we had an option to read a distil vector from disk, as opposed to computing it on the fly